### PR TITLE
Dockerfile: update golangci-lint v1.44.0

### PR DESCRIPTION
**Pull request by thaJeztah**
Looks like this may be needed for Go 1.18 https://github.com/moby/moby/pull/43194#issuecomment-1023720894

depends on:

- [x] https://github.com/moby/moby/pull/43200
- [x] https://github.com/moby/moby/pull/43202

### CODE REVIEW

File: hack/validate/golangci-lint.yml [hunk 1]
```diff
@@ -41,35 +41,44 @@ issues:
...
-      linters:
-        - gosec
-    - text: "G104: Errors unhandled"
+    # EXC0006
+    - text: "Use of unsafe calls should be audited"
       linters:
         - gosec
-    - text: "G204: Subprocess launch(ed with (variable|function call)|ing should be audited)"
+    # EXC0007
+    - text: "Subprocess launch(ed with variable|ing should be audited)"
```
Comment (thaJeztah) [inline hack/validate/golangci-lint.yml:67]: Looks like it doesn't properly match all cases; CI still fails on;

```
G204: Subprocess launched with a potential tainted input or cmd arguments
```

File: hack/validate/golangci-lint.yml [hunk 2]
```diff
@@ -41,35 +41,44 @@ issues:
...
+    # EXC0008
+    - text: "(G104|G307)"
       linters:
         - gosec
-    - text: "G304: Potential file inclusion via variable"
+    # EXC0009
+    - text: "(Expect directory permissions to be 0750 or less|Expect file permissions to be 0600 or less)"
       linters:
         - gosec
-    - text: "G306: Expect WriteFile permissions to be 0600 or less"
```
Comment (thaJeztah) [inline hack/validate/golangci-lint.yml:68]: New rule doesn't match all cases;

```
G306: Expect WriteFile permissions to be 0600 or less
```

**Comment by thaJeztah** [comment]
Windows failure is unrelated (known flaky: https://github.com/moby/moby/issues/43012)

```
=== RUN   TestDockerSuite/TestSlowStdinClosing
    docker_cli_run_test.go:4177: assertion failed: error is not nil: exit status 127
    --- FAIL: TestDockerSuite/TestSlowStdinClosing (8.60s)
```

**Comment by thaJeztah** [comment]
Another flaky (twice) on Windows `TestRunInteractiveWithRestartPolicy`; (tracked in https://github.com/moby/moby/issues/39352)

````
=== RUN   TestDockerSuite/TestRunInteractiveWithRestartPolicy
    docker_cli_run_test.go:1796: assertion failed: 
        Command:  d:\CI\PR-43195\6\binary\docker.exe run -i --name test-inter-restart --restart=always busybox sh
        ExitCode: 0
        Stdout:   
        Stderr:   
        
        Failures:
        ExitCode was 0 expected 11
    --- FAIL: TestDockerSuite/TestRunInteractiveWithRestartPolicy (5.67s)
```

**Review by tianon — APPROVED** [review]

**Pull request merged**